### PR TITLE
update Boost to 1.71.0

### PR DESCRIPTION
**Pull request by dan-42**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/update.html)
  step by step carefully. **YES**

* I've tested this package remotely and have excluded all broken builds. **NO**
  Here is the links to the Travis/AppVeyor with:
    * https://travis-ci.org/dan-42/hunter/builds/598490205
    * https://ci.appveyor.com/project/dan-42/hunter/builds/28140957

 * The boost package https://github.com/cpp-pm/boost/tree/hunter-1.71.0 was created by using the following script https://gist.github.com/dan-42/ef7ac7306dcde31efa97d66a227baf3e

### CODE REVIEW

File: cmake/projects/Boost/hunter.cmake [hunk 1]
```diff
@@ -148,6 +148,17 @@ hunter_add_version(
     e6bb97b5109c7c15ea459cf2b1a9d52cbf88a89e
 )
 
+hunter_add_version(
+    PACKAGE_NAME
+    Boost
+    VERSION
+    "1.71.0"
```
Comment (rbsheth) [inline cmake/projects/Boost/hunter.cmake]: Please match the version to the release. In this case, `1.71.0-p0`.

Comment (dan-42) [reply cmake/projects/Boost/hunter.cmake]: Thanks for pointing out, I will update the version to 1.71.0-p0

**Comment by dan-42** [comment]
Concerning the failed builds on AppVeyor, 
Boost 1.70.0-p0 did not run any tests, or at least any I could find.
It was added directly by @ruslo see https://github.com/ruslo/hunter/commits/master/cmake/projects/Boost/hunter.cmake

How ever the PR for 1.69.0-p1 by @Bjoe did already disable the 2 configs that failed here.

**Comment by bkotzz** [comment]
From here https://github.com/cpp-pm/hunter-testing/blob/pkg.boost/appveyor.yml, I think you’re right - it looks like the three windows toolchains failing is expected. But also looking at that build matrix (which came from ingenue/hunter), it looks like they normally test a bunch of the sub packages in CI as well. Should we be doing that (and same for Travis)?

**Comment by dan-42** [comment]
Yes, for now I think it makes sens to stick to the same way.
I'll update the PR soon this week.

Still if I find the time, I'll try looking into it why they break.

**Comment by xsacha** [comment]
> 'C:/Program' is not recognized as an internal or external command,
Would appear something isn't quoting the path.

If nmake/ninja/msys didn't work before, I say push this now and fix later. You can always make a revision to the scheme which fixes a bug such as this in a separate commit. It would likely fix all previous versions too.

**Comment by dan-42** [comment]
Sry got caught up in other stuff.

I Updated the requested version string from  ``` 1.71.0 ```  to ```1.71.0-p0```
I used the toolchains pointed out by @bkotzz 

builds running:
 * https://travis-ci.org/dan-42/hunter/builds/598490205
 * https://ci.appveyor.com/project/dan-42/hunter/builds/28140957

**Comment by dan-42** [comment]
So __all test pass__  with all sub components.
I think this can be merged, as it provides the latest version of boost and does not break any existing tool-chain.

But still would be nice if the broken jobs are fixed in the future.
I plan to look into it. But I cannot promise when i will find the time :-)

What I don't understand is why the appveyor build for this PR breaks.
It also points to @bkotzz `https://ci.appveyor.com/project/bkotzz/hunter-kif9a/builds/28140943`
and  not  to  my  instance `https://ci.appveyor.com/project/dan-42/hunter/builds/28147164`

Maybe because the appveyor project was not setup when I created this PR?
Or I did something else wrong.

**Comment by bkotzz** [comment]
Don’t worry about the builds that we’re already broken. And the appveyor on this repo isn’t set up correctly yet - it’s there for some work I’m doing to improve CI, but it’s not ready yet.

**Pull request merged**